src/pvrclient-argustv.cpp

```diff
@@ -1492,7 +1505,33 @@ bool cPVRClientArgusTV::FindRecEntryUNC(const std::string &recId, std::string &r
   if (iter == m_RecordingsMap.end())
     return false;
 
-  recEntryURL = ToUNC(iter->second.c_str());
+  recEntryURL = ToUNC(iter->second);
+  if (recEntryURL == "")
+    return false;
+
+  return true;
+}
+
+bool cPVRClientArgusTV::FindRecEntry(const std::string &recId, std::string &recEntryURL)
+{
+  auto iter = m_RecordingsMap.find(recId);
+  if (iter == m_RecordingsMap.end())
+    return false;
+
+  recEntryURL = iter->second;
+  if (!g_szUser.empty())
+  {
+    if (insertUser(recEntryURL))
+    {
+      XBMC->Log(LOG_DEBUG, "Account Info added to SMB url");
+    }
+    else
+    {
+      // recEntryURL does not contain "smb://"
+      XBMC->Log(LOG_DEBUG, "Account Info NOT added to SMB url");
+    }
+  }
```

Comment:
I have seen the above exact ~15 lines before in this diff... ;-) You should factor this out into a new private member function.